File: PAT/basic_level/1051_复数乘法/complex_number.py
from __future__ import print_function
import cmath


r1, p1, r2, p2 = map(float, raw_input().split())

z1 = cmath.rect(r1, p1)
z2 = cmath.rect(r2, p2)
z3 = z1*z2

# import math
# img = r1*r2*math.sin(p1+p2)
# real=r1*r2*math.cos(p1+p2)

img = z3.imag
real = z3.real

# 0.01 or 0.005
if abs(img)<0.01:
	img=0
if abs(real)<0.01:
	real=0

if img<0:
	print('%.2f-%.2fi' % (real, abs(img)))
else:
	print('%.2f+%.2fi' % (real, abs(img)))



'''
2.3 3.5 5.2 0.4


-8.68215035823-8.22568326384j

-8.68-8.23i

'''

File: PAT/basic_level/1051_复数乘法/complex_number.cpp
#include <iostream>
#include <cstdio>
#include <complex>
#include <cmath>
using namespace std;

int main(){
	double r1, p1, r2, p2;
	scanf("%lf %lf %lf %lf", &r1, &p1, &r2, &p2);
	double real1 = r1 * cos(p1);
	double imag1 = r1 * sin(p1);
	double real2 = r2 * cos(p2);
	double imag2 = r2 * sin(p2);
	complex<double> a(real1, imag1), b(real2, imag2);
	complex<double> result(a*b);
	double real = result.real();
	double imag = result.imag();

	if( fabs(imag)<0.01 )
		imag = 0;
	if( fabs(real)<0.01 )
		real = 0;

	if(imag<0)
		printf( "%.2lf-%.2lfi\n", real, fabs(imag) );
	else
		printf( "%.2lf+%.2lfi\n", real, fabs(imag) );

	return 0;
}
